kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/test/ExpectFailureException.kt

```diff
@@ -0,0 +1,10 @@
+package io.kotest.core.test
+
+/**
+ * This exception is thrown when a test fails an expect block.
+ */
+object ExpectFailureException : Exception()
+
+fun TestScope.runIf(thunk: () -> Boolean) {
```

Comment:
I think we could make this `runIf(condition: Boolean)`, then we can have rhe one taking a function return `T`